codec/reflectcodec/type_codec.go

```diff
@@ -113,12 +118,22 @@ func (c *genericCodec) size(value reflect.Value) (int, bool, error) {
 	case reflect.String:
 		return wrappers.StringLen(value.String()), false, nil
 	case reflect.Ptr:
-		if value.IsNil() {
-			// Can't marshal nil pointers (but nil slices are fine)
+		if omitEmpty {
+			if value.IsNil() {
+				return wrappers.BoolLen, false, nil
+			}
+			size, _, err := c.size(value.Elem(), omitEmpty)
```

Comment:
I'm a bit torn on how we are passing through `omitEmpty`. For the `maxSliceLen` we don't pass the custom value until hitting a different struct. I feel like they should have the same behavior

Reply:
I think 708037722 does exactly that and reduces greatly the complexity added by this PR. What are your thoughts @StephenButtolph ?